pkg/apps/gitops.go

```diff
@@ -42,6 +42,16 @@ func (o *GitOpsOptions) AddApp(app string, dir string, version string, repositor
 		GitProvider: o.GitProvider,
 	}
 
+	prDir := o.EnvironmentsDir
```

Comment:
I find it very confusing that parts of the required input for this function come via function parameters and others by accessing the `GitOpsOptions` instance. For example, `autoMerge` is passed as a parameter,  `o.EnvironmentsDir` not.

Reply:
I've tidied it up a bit, but it's still kinda janky.